vic/drivers/shared_image/include/vic_mpi.h

```diff
@@ -32,6 +32,14 @@
 
 #define VIC_MPI_ROOT 0
 
+/******************************************************************************
+ *  * @brief   This structure stores netCDF file name and corresponding nc_id
+ *   *****************************************************************************/
+typedef struct {
+    char    nc_filename[MAXSTRING];
+    int     nc_id;
+} nameid_struct;
+
```

Comment:
That doesn't quite work either since some of the netcdf function require knowledge of the `domain_struct`. So instead I think it is better to move it one level up into `vic_driver_shared_image.h`, since at least that will isolate the `mpi` related functions into `vic_mpi.h` (I know that goes against what I suggested earlier)